### ISSUE TITLE
Desktop Client 3.0.1 Windows constant high CPU load ~20-32% (while doing nothing)

### ISSUE THREAD

**Issue by jonathanmmm** [nextcloud/desktop#2379]
Like in https://github.com/nextcloud/desktop/issues/964
I tried the trick with the registration and it seemed to work, but it didn't while writing a post there the CPU usage goes up again.

My Machine Windows 10 Laptop (i5-6300U, 16GB RAM, Windows 10 1909, SSD)

I have around 25% CPU load, got down and up (Task Manager showed that Nextcloud drained a lot of power, so my Laptops battery goos down quickly, too).

I am syncing 301 GB and 144k files (through Windows Explorer checking the folder).

I am just browsing the web. Have no documents or anything inside the Nextcloud folder open.

I am probably reinstalling 2.3.3 revision 1 again.
As sadly new features don't help me if all programs lack and I feel like I have an ten year old laptop and have to kill nextcloud for using the pc.

Hope this gets fixed, I keep 3.0.1 on for a few days, so if I can provide log files, I will be happy to do so.

**Comment by er-vin** [comment]
Could be interesting if you start the client with the --logdebug and --logfile parameters and see if it keeps claiming something in the logs while doing nothing from your point of view.

Note we spotted a performance regression last week, should be fixed in 3.0.2 at the end of the month.

**Comment by jonathanmmm** [comment]
Ok, I will enable them and start the client.

I believe this bug was there before 3.x version (so also in like 2.5 or 2.6). But can't say when it started.

Will report back with logfiles if something is in there that is interesting.

**Comment by jonathanmmm** [comment]
High, I got an log file.
Ran it like for 6 minutes. File is 344MB big.
Has 1.2 million lines.
I tried to replace every domain name and file name with generic names but it takes too long and I can't know if I found all. Is there a a way to automatically anonymize the log file?

I can see a lot of readings of files and database reading, like "Database entry found"
Then a lot:

**Comment by er-vin** [comment]
So it looks like there's a sync on-going. It could be that it takes forever to complete.

Please try again with 3.0.2 when it is out and let us know, it might go back to normal then (an index is missing in the database which made things unfortunately longer than needed).

**Comment by x-jokay** [comment]
Can confirm this for `3.0.2` as well.

**Comment by Michitroniker** [comment]
Hi there,
I encountered probably the same issue, with constant cpu load of about 25%, today for the first time.
After stop the process and start the client again it seems to be over. I will observe it furthermore.

I'm using the Desktop Client 3.0.2 on a Win10 Pro 1909

I will provide a log as soon i have one.

**Comment by Sebaristoteles** [comment]
Similar issue with Desktop Client 3.1.1 and Windows 10 Pro version 20H2, build 19042.685.
Task Manager shows 15% CPU of a Ryzen 4800H at 2 am in the morning.
More suspicious is that when I "Exit Nextcloud" on the taskbar, there is still the task running in the task manager. And after the restart of the Desktop Client there are two instances of Nextcloud running in the Task Manager (see link below for screenshot). The CPU usage stays that high even when the Desktop Client is in sync with the NC.

I had a look at the properties of the process and it does not lead to "nextcloud.exe" as the client properties show (unfortunately I made no screenshot of the properties).

Is there any information about an NC virus or Crypto Miner?

[https://drive.google.com/file/d/1nR9wtfHmu82hy-Ut4XZYT5a_q2tXVO43/view?usp=sharing](url)

**Comment by kveys** [comment]
Hi,
I am observing the same issue here. Average CPU utilization around 35%. Intel i5, 8Gb ram.
Mine if a very small NC test installation. (around 100 files, I'm guessing) I ran procmon for 3 minutes and the only write activity I could see, were write actions to the logfile. The rest were reading some dll's and SSL libraries. 
Should you want to investigate further, I can share the NC link. 
Kind regs

**Comment by johannes314** [comment]
same issue here. 20-25% CPU usage and Task Manager says very high power consumption. I am running NC Client  3.1.2
I also tried to locate the source file behind th task and it points to a file f9f0.rbf in the c:\config.msi folder. i googled that and it has something to do with role ack scripts to for when i would uninstall NC.

Any suugestions?

**Comment by Herve-M** [comment]
I had the same issue, only after updating. 
It's actually a script `3d84d2fc.rbf` ran by `config.msi` as @johannes314 state. 
I made a full dump of the main process, if any dev. is interrested.

**Comment by Herve-M** [comment]
Same problem as described by @johannes314 apart that the issue only occured after upgrading. After restarting Windows, no problem anymore. 
I have made a process dump of the "nextcloud" process if any dev. is interrested.

**Comment by tilllt** [comment]
Same here, this is really not acceptable. My laptop constantly goes to leaf-blower mode because of the Nextcloud Client.
![Screenshot_22](https://user-images.githubusercontent.com/1854364/109431601-b5e55f00-7a07-11eb-907e-f2c1c5fec701.jpg)

**Comment by tilllt** [comment]
BTW could somebody escalate this issue? Just imagining how much energy is wasted because of thousands of computers running in constant 30$ CPU load for nothing, because of a software bug, is kind of shocking and probably not in line with the ethics of nextcloud.

**Comment by andreipoe** [comment]
This seems to be triggered only after a client update. The updater spawns another process shown as `Nextcloud` in the task manager, which doesn't stop if you exit the Nextcloud client. Manually stopping the troublesome process makes it go away, and it doesn't seem to come back with normal use of the client.

**Comment by tilllt** [comment]
I stopped and restarted the client several times after updating and it definitely did not go away, so cannot confirm.

**Comment by Sebaristoteles** [comment]
Hello @tilllt, you are right, closing the client is not sufficient. 
But as @andreipoe and I some posts earlier have pointed out: There is an extra process for Nextcloud in the Task Manager that you can stop manually. Just look in the long list of processes in the Task Manager for the process that has a high CPU load and is also called Nextcloud (if I remember correctly).
For sure, this is just an individual workaround and it would be nice if the Team from Desktop Client could have a look into this.

**Comment by tilllt** [comment]
There is no extra process in my task manager related to NC - or it's name is completely untelated to NC and thus impossible to find (in the 100s of processes running on my machine). I will gladly try to stop if if somebody points me to the name. In process-explorer it looks to me as if the culprit is a QT5 related task but maybe i am wrong.

Just as an anecdote: my i7-7700HQ has a TDP of 45W, so JUST the CPU running at 30% will consume roughly 15W.
FOR NOTHING. And this is a pretty power efficient machine. Just imagine if on a worldwide level, people have their CPU's running at 30% load, thats thousands of Watts blown out of the window.

I think programmers are sometimes unaware of this type of repercusssions their work has.

**Comment by FlexW** [comment]
@tilllt I'm unable to reproduce this issue on my side for now. Could please have a look at your Nextcloud installation directory if you have a folder named `bearer` with a file named `qgenerbearer.dll` in it? I ask because we had a report from a guy on OSX that had also high cpu usage and it was caused by this dll.
From the newer installs, we remove this dll but maybe for some reason it is still left.

**Comment by FlexW** [comment]
> Is there any information about an NC virus or Crypto Miner?

I just want to be clear on that: There is no virus or crypto-miner embedded into the client. The code is as you can see in the public domain and you can verify that yourself:)

**Comment by morbificagent** [comment]
@andreipoe Same here. Killing this process helps for the moment and i cant see a problem after that.

**Comment by tilllt** [comment]
I solved the issue by completely uninstalling and reinstalling the Nextcloud Client. There was no "leftover" process running on my system at least nothing that i could identify in task manager / process explorer. In any case, it is something that the Client Team should probably investigate further. As said before, it is not acceptable that an idle process takes over 30% cpu. Manually killing a process in the task manager is merely a bad workaround but not a solution.

**Comment by FlexW** [comment]
@tilllt Was the bearer folder there before you uninstalled? It's sometimes really difficult to investigate further if you are not able to reproduce the problem and indeed it could be the case that the issues described here have different origins.

**Comment by tilllt** [comment]
@FlexW I already had reinstalled the Client when you asked about the existence of the qgenerbearer.dll.

**Comment by webfischi** [comment]
I have between 15-20% CPU usage with an i7 4770k only syncing 120 MB with very rare changes.

**Comment by HoneypotFX** [comment]
I have a similar issue. 50-60% CPU usage with an i7-5600U with nothing going on.

**Comment by mgallien** [comment]
@corpetty @HoneypotFX Can you please open a new issue and try to provide logs as explained in the new issue template ?
This issue is old and I cannot be sure your problem is similar to the one initially reported here.

**Comment by corpetty** [comment]
@mgallien I have restarted the VM and so far it is good. If I can reproduce the issue consistently I will make a new issue thread.

**Comment by webfischi** [comment]
I'm not quite sure if it is different, syncing a file constantly over a period of time without restarting the PC, and then let it idle a few days, the CPU usage went up to 20-32% again

**Comment by thymaro** [comment]
Ping! 

- Same thing happened to me after updating from v3.2.1 to v3.2.2. 
- I have a suspicion (after reading this thread and noticing others have reported having AMD Ryzen) this could have something to do with AMD Ryzen, I have a pro 7. 
- I closed the sync client and nothing changed. The other process was still visible in Task manager.
- After closing the suspicious task some in here were talking about, cpu usage decreased from 25% to 2% and the helicopter whirring around in the room landed and I think the pilot went for a coffee break.
- Started the client again and the pilot is still on her break.
- All seems well. For now.

**Comment by webfischi** [comment]
This is not a Ryzen issue, i have an Intel Core i7 4770

**Comment by JohannesDeml** [comment]
@thymaro Hm, I can't confirm the AMD Ryzen theory. I hadn't had the problem with my desktop PC (Using an AMD Ryzen), but with my Lenovo Carbon X1 4th gen. I kept running Nextcloud for 30 minutes, but the CPU usage didn't stop. Restarting the client fixed the problem for me.

**Comment by thymaro** [comment]
Maybe this can help further analyze the issue. I didn't translate the German parts because I think it's clear enough like it is.

```
Betriebssystemversion:                         10.0.19042 Nicht zutreffend Build 19042
Systemhersteller:                              LENOVO
Systemmodell:                                  20Y1CTO1WW
Systemtyp:                                     x64-based PC
Prozessor(en):                                 1 Prozessor(en) installiert.
                                               [01]: AMD64 Family 23 Model 96 Stepping 1 AuthenticAMD ~1700 MHz
BIOS-Version:                                  LENOVO R1BET61W(1.30 ), 21.12.2020
Gesamter physischer Speicher:                  16 GB
Hotfix(es):                                    10 Hotfix(e) installiert.
                                               [01]: KB4601554
                                               [02]: KB4534170
                                               [03]: KB4537759
                                               [04]: KB4545706
                                               [05]: KB4562830
                                               [06]: KB4577266
                                               [07]: KB4577586
                                               [08]: KB4589212
                                               [09]: KB5003173
                                               [10]: KB5003242
```

**Comment by adymorz** [comment]
I had the same issue.

After I closed the nextcloud client over the system tray, the process was still running and generating about 20 % CPU load. I stopped the process and right after, I noticed a windows telemetry service using a lot of CPU for some seconds.

I then decided, to disable telemetry and stop the connected user experiences and telemetry service, because I do not want those to be active anyways.

Since disabling the telemetry services, the high load of the nextcloud client is gone.

You want to give it a try? Go to https://www.thewindowsclub.com/windows-10-telemetry and go to chapter "Disable Telemetry".

**Comment by FireEmerald** [comment]
This issue is **not** resolved, it still happens after updating to 3.2.2 on my system.

**Comment by abaart** [comment]
Same issue here....

**Comment by ossimantylahti** [comment]
Hi. I would also like to report that this is happening still. The sync is finished, but Nextcloud Windows client generates about 25 % constant processor load and power consumption is very high according to Windows task manager.

**Comment by adymorz** [comment]
Could someone just try out [disabling telemetry as I described June the 3d](https://github.com/nextcloud/desktop/issues/2379#issuecomment-853778782)?

**Comment by stardust342** [comment]
Same issue after updating 3.2.1 to 3.2.2 on Ryzen 5600x and an E5-2637 v2 (virtualized) so this seems to be platform agnostic. 
Killing via task manager resolved the issue on both systems.

**Comment by ifaist0s** [comment]
I have just updated to Nextcloud Desktop client v3.2.3, on Windows 10 Pro 21H1. In Task Manager (TM for short), there is a leftover process called Nextcloud with a blue icon, which is causing high CPU usage, and the regular Nextcloud process of the sync client with the green icon. Here is a screenshot for a visual representation (first column is the % CPU Usage which goes around 23-30%):
![image](https://user-images.githubusercontent.com/998239/124346176-0fdc7800-dbe6-11eb-9462-e545fffb40d8.png)

I have been having this issue since +8 months (maybe more, I'm not sure) on my desktop PC (Intel(R) Core(TM) i5-8600K, 16GB RAM) and on my laptop (AMD Ryzen) for about 6+ months. This is happening only and always while updating the client and I need to remember to kill the second process, after the update is complete.

Digging some more with the TM, here are the two processes (in the Details tab in TM).
![image](https://user-images.githubusercontent.com/998239/124346780-68614480-dbe9-11eb-8d10-87315e812313.png)
The highlighted process is the one that is causing the CPU load. Right-clicking that process and choosing Properties, I get this `20b09.rbf` file which bears a valid signature from Nextcloud GmbH (signer name SwissSign ZertES TSA UNIT CH-2018, date May 27th 2021 15:55:20):
![image](https://user-images.githubusercontent.com/998239/124346796-89299a00-dbe9-11eb-9d1c-316d5a3ef7e6.png)
which I have no idea what it is. Guessing from the location (C:\Config.Msi) I'd say it has something to do with [Windows installer](https://superuser.com/a/8050).

As far as I remember, Nextcloud Client Installation method changed at some point to Windows Installer from NSIS(?). I personally don't like Windows Installer because I have faced many problems with it over the years. But I don't question the team's decision to stop using NSIS and build on Windows Installer instead. Maybe it would be worth trying (if possible) with an NSIS installer/updater next time (i.e. next update) instead of the regular Windows Installer method. If the update goes all and well, we may have pinpointed the problem.

@FlexW I have neither the `bearer` nor the `qgenerbearer.dll`.

I understand that this issue is marked as closed, but since it seems that it IS an **ongoing issue** for some people, I'm posting my findings with the hope to help identify the underlying problem.

**Comment by sbw** [comment]
> I have just updated to Nextcloud Desktop client v3.2.3 ... there is a leftover process called Nextcloud with a blue icon, which is causing high CPU usage, and the regular Nextcloud process of the sync client with the green icon ...

I've seen the same behavior after each update for months.  I was surprised to see this issue got closed.  I hope it will get some attention.

**Comment by PopoWow** [comment]
On my machine most definitely not resolved.  In the past, using the .EXE installer helped IIRC but it looks like that's no longer available.

Machine info: i3- 4010U.  Windows 20H1 x64, 4GB.  Definitely not a power machine but I just use it for remoting in.  The constant 25% CPU usage brings this oldster to a crawl.

Nextcloud client: 3.2.3.20210624

I can try and get logs if that would help.

**Comment by Sebaristoteles** [comment]
I can provide logs, but will not post them here. Please contact me to get them.
Moreover, I let the process running for now. So, please contact me if you like to troubleshoot with an example.

I can confirm that it seems to be related to the installing process (on Windows)
<img width="491" alt="Nextcloud_installer_issue" src="https://user-images.githubusercontent.com/56370298/124760113-b00d0680-df30-11eb-874e-1d329672a331.png">

<img width="258" alt="properties_process" src="https://user-images.githubusercontent.com/56370298/124760545-23af1380-df31-11eb-9e33-fce7decb7131.png">

**Comment by brneor** [comment]
I can confirm it still happens on version 3.2.3 on Windows exactly as stated above.

**Comment by Discostu36** [comment]
Same issue here. High CPU load after update. Closed NC client, killed the process and opened the NC client again, and it's gone. Why doesn't this get reopened?

**Comment by claell** [comment]
Looks like this has never been resolved, correct? Reopening for now.

**Comment by causenet** [comment]
Same issue here. Updated and CPU load ~ 30%.

**Comment by Snuupy** [comment]
I have 2 clients, both running latest windows 10, intel-6300 and ryzen 7 4750u, both had this issue today.

**Comment by allexzander** [comment]
@Sebaristoteles Could you upload logs here? https://cloud.nextcloud.com/s/CzJtnEZKi9H45n9

Also, if you right-click and select "Properties", when the popup opens, could you take a screenshot of "General" and "Details" tab for each Nextcloud process? If you don't want to take screenshots, can you write here what are the version and path for both .exe files?

**Comment by allexzander** [comment]
@Sebaristoteles Update. I can see, the first process is somehow related to the MSI installer. Does it always happen when you update the client? If you remove it via Program Files, make sure it is not running, then, install the previous version from here https://download.nextcloud.com/desktop/releases/Windows/ and then - auto-update as before, will it also be the case that 2 processes are running?

**Comment by allexzander** [comment]
@Sebaristoteles Another important question that you may want to answer to help me find the root cause - what are the users running both processes? I mean, one user can be - your username, and another one - someone else, like - system? Can you also check that in the Task Manager and share with me?

**Comment by vedran-dobos** [comment]
Hello @allexzander

Not trying to highjack the thread, but I just stumbled upon an instance of the same issue. So here are a few screenshots requested.
The process seems to be running under the same user.

As it's a somewhat less important pc, I'll keep the offending process running at the moment.
I also experienced this problem on a multitude of occasions, so would like to assist in solving it.
Thanks!


![ss1](https://user-images.githubusercontent.com/19259909/125907553-b149af02-2f43-4e95-bf66-c503102f8076.png)
![ss2](https://user-images.githubusercontent.com/19259909/125907557-db522f09-a940-49d3-93d3-80cee1867387.png)
![ss3](https://user-images.githubusercontent.com/19259909/125907560-e6ca2db4-22a3-4195-9fdf-17a310f6182e.png)
![ss4](https://user-images.githubusercontent.com/19259909/125907561-6827488f-e7a3-4be1-8563-c2a76f4f8ff3.png)
![ss5](https://user-images.githubusercontent.com/19259909/125907563-750c6c5e-86e4-4790-83dc-792bea4bf5a8.png)

**Comment by allexzander** [comment]
@vedran-dobos Thank you. Interesting. Do you remember by any chance if you always have that Config.msi folder on your drive and if every program is using it when installing/updating. I am trying to find out if there is any configuration in Windows that I can change to achieve the same behavior on my local machine.

**Comment by vedran-dobos** [comment]
@allexzander Hmm... Cannot say for sure, but I seem to remember seeing it when a(ny) program is installed or updated. Although, could be installer dependent.
I browsed around, and I see the folder still exists for the PCs that I know had the issue after upgrading to version 3.2.4. There is 1 up to a few dozens files ending in .rbf inside. The date on the files does not correspond to the date the problem occurred (maybe in the past, but really cannot say).

Anything else I can check?


![ss6](https://user-images.githubusercontent.com/19259909/125915685-c3310904-0d87-4142-99aa-7bde3c2e537e.png)

**Comment by jonathanmmm** [comment]
This problem was resolved after I put an older image from my Laptop onto it (windpws Update destroyed my operating system, dont click optional updates). On the old image was also nextcloud client (was like a junp in time back a year). Until now I don't have this 20-30% problem. But it id just inresponsive, cant use the tool, it is laggy or freezes. The whole new UI makes it really difficult to use. Even rightclick takes a long time.

**Comment by Sebaristoteles** [comment]
@allexzander Thanks for going after this!
I uploaded the log files. Not sure if these help in this issue. I think it is somehow related to the Windows Installer (@ifaist0s also suggested this above).

I could not check the things you mentioned, because after a Windows Update with a restart the second Nextcloud instance with the link to the Windows Installer was not there anymore.

I followed your procedure trying to reproduce the issue by uninstalling, rebooting, installing 3.2.3, rebooting, starting Nextcloud, letting the automatic update do its work. The issue did not pop up this time, but I can definitely confirm that I had it after the updates 3.2.2->3.2.3 and 3.2.3->3.24. So, it happened twice within 2-3 weeks.

What I noticed is that (if I remember correctly), the updating to 3.2.4 did *not* ask in the windows user control window of the rights to install as it did for the installation of 3.2.3. This might be completely unrelated, but maybe these rights are the prerequisite, while on the other hand it probably did not ask for it because I already granted these rights to Nextcloud some minutes before.
![windows_user_control_blacked](https://user-images.githubusercontent.com/56370298/125941826-170cc092-541b-48ba-bb34-9e8e579f422a.png)

**Comment by allexzander** [comment]
Keep in mind, the fix will start to work in 3.3.1. The bug is in the client's updater code, so, for it to start working properly, you must first install 3.3.0 (which is going to be released soon), then, update to 3.3.1 and subsequent releases should work properly, without the old process sitting there and eating your CPU cycles. After updating to 3.3.0, please restart, to avoid the issue. 3.3.1 should then work fine without need to restart.

**Comment by uerseli** [comment]
Same issue here and struggeling with the problem for at least a year. I have de- & reinstalled Nextcloud without any improvement on the high CPU load. For me, the problem always occurs when an update is pending or something needs to be synchronized. The synchronization runs extremely slowly and chews up CPU, although I can rule out bandwidth problems. Right-clicking on the Nextcloud icon does nothing and kills the process. If anyone can help, I am NOT an expert, just a user and bloody beginner.

**Comment by pcismejr** [comment]
I've been struggling with this issue for several days now on 2 computers. It seemed to begin after a automatic update to 3.8.0  . I've also recently updated my Nextcloud server to AIO. The problem progressively got worse until my Windows 11 laptop would not sync at all. The client would instantly go unresponsive a few seconds after starting up. I also got the high CPU condition. I resolved the problem by uninstalling the client and then manually deleting the folder found in C:\Users\myprofile\AppData\Roaming\Nextcloud  . I then reinstalled 3.8.0 and logged in and synced. No more unresponsive client. I'll monitor for a few days and if it reoccurs, give an update. Hope this helps someone.
